### PR TITLE
Feature/integrate react hook form with userinput

### LINKED ISSUE #112: Integrate React-hook-form with userInput UI
**Is blocked by #108
**Is Related to #85
**Acceptance Criteria
1. Implement react-hook-form
2. User functionality still works
3. Validation works with zod implementation related to #85

### PR DESCRIPTION
**Pull request by lomeintheory**
## Description
- Implemented UserInput validation with react-hook-form and the UserInputValidationSchema created in #85.
- Utilized react-hook-form functions to centralize form state management and optimize performance for UserInputBar and UserInputCheckbox (sidenote: although it's not within the scope of this ticket, I created a UserInputCheckbox component to maintain consistency with the other components).
- Wrapped our components in a form element so pressing Enter on our keyboard will trigger the form submit now.

Fixes #112 

## Type of change

- [x] ✨  New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Ran on localhost to confirm that form submission is prevented when our validation schema isn't followed. Form errors are appropriately shown when input is less than 3 characters or more than 100 characters. 

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [ ] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: client/components/prompt-input/PromptInputForm.tsx [hunk 1]
```diff
@@ -84,18 +106,8 @@ const PromptInputForm = () => {
...
-
-  const handleSubmit = () => {
-    newUserInputMutation.mutate({ input: inputValue });
-  };
-
-  const handleCheckboxClick = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const newCheckedState = event.target.checked;
-    setIsChecked(newCheckedState);
-    localStorage.setItem('isChecked', newCheckedState.toString());
+  const onSubmit = (userInput: UserInputPayload) => {
```
Comment (colinvandergraaf) [inline client/components/prompt-input/PromptInputForm.tsx:109]: I think we might need a `reset()` int he onSuccess handler to clear the form state after submssion.

File: client/components/prompt-input/PromptInputForm.tsx [hunk 2]
```diff
@@ -111,36 +123,49 @@ const PromptInputForm = () => {
...
+                    {...field}
+                    checked={field.value}
+                  />
+                )}
+              />
+            }
+            label="Content on this site may be generated by artifical intelligence (AI). Please use discretion and verify information where necessary."
+            sx={{ color: 'text.primary' }}
+          />
+          <GenerateButton
```
Comment (colinvandergraaf) [inline client/components/prompt-input/PromptInputForm.tsx:160]: Are handling loading states in the UI part of this ticket?
Also it looks like the Figma has a gray color for the checkbox rather thant the green seen here. Not sure if this part of this PR.

File: client/components/prompt-input/PromptInputForm.tsx (hunk 1 above)
Comment (lomeintheory) [reply client/components/prompt-input/PromptInputForm.tsx:109]: Good catch. I also added an isSubmitted value to the GenerateButton's disabled prop so inputs can't be submitted after the first submission. Makes for good UX at least until I finish the Loading state UI.

File: client/components/prompt-input/PromptInputForm.tsx (hunk 2 above)
Comment (lomeintheory) [reply client/components/prompt-input/PromptInputForm.tsx:160]: The Loading/Submitting states are part of a separate ticket that I'm in the process of implementing. As for the checkbox, I must've set the color incorrectly, although it was never faithful to the Figma design to begin with. I am planning on assigning myself a ticket to style the UserInput UI the way it's presented in the updated Figma design.

Comment (colinvandergraaf) [reply client/components/prompt-input/PromptInputForm.tsx:162]: 👍

File: client/components/prompt-input/PromptInputForm.tsx [hunk 3]
```diff
@@ -44,17 +47,37 @@ const GenerateButton = styled(Button)<ButtonProps>(({ theme }) => ({
...
+    handleSubmit,
+    reset,
+    control,
+    watch,
+    formState: { errors, isSubmitted },
+  } = useForm({
+    resolver: zodResolver(userInputValidationSchema),
+    defaultValues: {
+      input: '',
+      isChecked: JSON.parse(localStorage.getItem('isChecked') || 'false'),
```
Comment (danielboisselle) [inline client/components/prompt-input/PromptInputForm.tsx:67]: Could you change this to

`isChecked: Boolean(localStorage.getItem('isChecked')) || false,`

You can add this to a followup pull request

**Review by danielboisselle — APPROVED** [review]

**Review by colinvandergraaf — APPROVED** [review]

**Pull request merged**